src/main/java/com/artipie/asto/memory/InMemoryStorage.java

```diff
@@ -42,7 +44,15 @@
      * Ctor.
      */
     public InMemoryStorage() {
-        this.data = new TreeMap<>();
+        this(Collections.emptyMap());
+    }
+
+    /**
+     * Ctor.
+     * @param data Content of storage
+     */
+    public InMemoryStorage(final Map<String, byte[]> data) {
+        this.data = new TreeMap<>(data);
```

Comment:
This way we can't access passed data by reference: `new TreeMap` calls `putAll()` method on constructor parameter passed, so it just copies all entries from `data` param to internal map. I'd make this constructor package-private and require `NavigableMap<>` isntead of `Map<>` as param, and assign it directly to class field - this way we will be able to access it directly from benchmark storage without sync methods call.